File: fib.cs
using System;
using System.Diagnostics;

class Program
{
    static long Fibonacci(int n)
    {
        if (n <= 1) return n;
        long a = 0, b = 1, temp;
        for (int i = 2; i <= n; i++)
        {
            temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }

    static void Main()
    {
        var stopwatch = Stopwatch.StartNew();
        long result = Fibonacci(50);
        stopwatch.Stop();
        
        long memoryUsed = GC.GetTotalMemory(false) / 1024;
        
        Console.WriteLine("Language: C#");
        Console.WriteLine($"Time: {stopwatch.Elapsed.TotalSeconds:F6} seconds");
        Console.WriteLine($"Memory: {memoryUsed} KB");
        Console.WriteLine($"Result: {result}");
    }
}

File: fib.rs
use std::time::Instant;

fn fibonacci(n: u32) -> u128 {
    if n <= 1 {
        return n as u128;
    }
    let mut a: u128 = 0;
    let mut b: u128 = 1;
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    let start = Instant::now();
    let result = fibonacci(100);
    let duration = start.elapsed();

    // Memory usage - approximate via allocation tracking
    let memory_kb = std::mem::size_of_val(&result) +
                    std::mem::size_of::<u128>() * 2;

    println!("Language: Rust");
    println!("Time: {:.6} seconds", duration.as_secs_f64());
    println!("Memory: {} KB", memory_kb / 1024);
    println!("Result: {}", result);
}
